File: core/src/main/java/com/lsxiao/loki/core/Loki.java
package com.lsxiao.loki.core;

import java.lang.reflect.Method;

/**
 * write with LokiRule
 * author:lsxiao
 * date:2017-05-25 21:55
 * github:https://github.com/lsxiao
 * zhihu:https://zhihu.com/people/lsxiao
 */

public class Loki {
    public static ValidateResult validate(Object object) {
        try {
            Class<?> lokiValidatorClass = Class.forName("com.lsxiao.loki.core.LokiValidator");
            Object validator = lokiValidatorClass.newInstance();
            Method validateMethod = lokiValidatorClass.getMethod("validate", Object.class);
            validateMethod.invoke(validator, object);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return (ValidateResult) new Object();
    }
}

File: core/src/main/java/com/lsxiao/loki/core/Loki.kt
package com.lsxiao.loki.core

/**
 * write with LokiRule
 * author:lsxiao
 * date:2017-05-25 21:55
 * github:https://github.com/lsxiao
 * zhihu:https://zhihu.com/people/lsxiao
 */

class Loki {
    companion object {
        private val validator: Validator by lazy {
            Class.forName("com.lsxiao.loki.core.LokiValidator").newInstance() as Validator
        }

        @JvmStatic
        fun check(any: Any): ValidateResult {
            validator.check(any)
            return Any() as ValidateResult
        }
    }
}
